scriptler/checkNodesLauncherVersion.groovy

```diff
@@ -12,19 +12,32 @@ import hudson.remoting.Launcher
 import hudson.slaves.SlaveComputer
 import jenkins.model.Jenkins
 
-def expectedVersion = Launcher.VERSION
-println "Expected Version = ${expectedVersion}"
+def expectedAgentVersion = Launcher.VERSION
+def expectedJavaVersion = System.getProperty("java.version")
+println "Master"
+println " Expected Agent Version = '${expectedAgentVersion}'"
+println " Expected Java Version = '${expectedJavaVersion}'"
 Jenkins.instance.getComputers()
         .findAll { it instanceof SlaveComputer }
         .each { computer ->
+    println "Node '${computer.name}'"
     if (!computer.getChannel()) {
-        println "Node ${computer.name} disconnected.."
+        println " is disconnected."
     } else {
-        def version = computer.getSlaveVersion()
-        if (!expectedVersion.equals(version)) {
-            println "${computer.name} - expected ${expectedVersion} but got ${version}"
-        } else {
-            println "${computer.name} - OK"
+        def isOk = true
+        def agentVersion = computer.getSlaveVersion()
+        if (!expectedAgentVersion.equals(agentVersion)) {
```

Comment:
I think checking on the major version should suffice to keep it simple. I can just split and compare accordingly.